stac_fastapi/tests/resources/test_collection.py

```diff
@@ -107,6 +107,7 @@ async def test_returns_valid_collection(ctx, app_client):
     collection.validate()
 
 
+@pytest.mark.skip(reason="Broken as of stac-fastapi v2.5.5")
```

Comment:
Is this fine to skip? Or do we need a new test?

Reply:
I did make an issue for this. I am not too sure about collection extensions and if people are using them. Ideally we would understand why this test started failing. If no one knows why it's failing, it should be fine to skip it for now and have it referenced as an issue I think.

Reply:
I added a new test. It's working with adding a collection but not updating one.

Reply:
I went through this with the debugger and it seems like Pydantic is omitting the field when validating the request model (`stac_fastapi.types.stac.Collection`), only retaining the defined fields.
https://github.com/tiangolo/fastapi/blob/0.98.0/fastapi/dependencies/utils.py#L785

So I'd expect this to happen on all request models, but we just don't notice it on Items objects because all extra fields are defined inside the properties field.

Reply:
It makes sense but when I POST a collection and then GET the collection after, the `iten_assets` field is still there? It's only with PUT that it disappears?

Reply:
I made an issue [here](https://github.com/stac-utils/stac-fastapi-elasticsearch-opensearch/issues/238) - hopefully this can be straightened out with the upcoming stac-fastpapi v3.0.0

Reply:
#238